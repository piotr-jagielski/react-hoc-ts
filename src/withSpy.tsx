import * as React from 'react';
import './App.css';

export const withSpy = <TProps extends object>(Inner: React.ComponentType<TProps>) =>
  class WithSpy extends React.Component<TProps> {
    public render() {
      return (
        <div>
          <Inner {...this.props} />
          <span>
          {JSON.stringify(this.props, undefined, 2)}
          </span>
        </div>
      );
    }
  }

export default withSpy;
