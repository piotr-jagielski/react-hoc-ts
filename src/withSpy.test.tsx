import { shallow } from 'enzyme';
import * as React from 'react';
import withSpy from './withSpy';

it('renders', () => {
  const Component = withSpy(Inner);

  const component = shallow(<Component a="a" b="b" c="c" />);

  expect(component.exists()).toBe(true);
});

it('renders inner component', () => {
  const Component = withSpy(Inner);

  const component = shallow(<Component a="a" b="b" c="c" />);

  const inner = component.find(Inner);

  expect(inner.exists()).toBe(true);
});

it('passes props to inner component', () => {
  const Component = withSpy(Inner);

  const component = shallow(<Component a="a" b="b" c="c" />);

  const inner = component.find(Inner);

  expect(inner.props()).toMatchObject({
    a: 'a',
    b: 'b',
    c: 'c',
  })
});

it('renders stringified props', () => {
  const Component = withSpy(Inner);

  const component = shallow(<Component a="a" b="b" c="c" />);

  const text = component.find('span').text();

  expect(text).toMatchSnapshot();
});

interface InnerProps {
  a: string;
  b: string;
  c: string;
}

const Inner = (props: InnerProps) => null;