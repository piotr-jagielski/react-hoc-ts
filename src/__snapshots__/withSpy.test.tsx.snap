// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders stringified props 1`] = `
"{
  \\"a\\": \\"a\\",
  \\"b\\": \\"b\\",
  \\"c\\": \\"c\\"
}"
`;
